nucypher/policy/collections.py

```diff
@@ -209,6 +211,67 @@ def __len__(self):
         return len(self.destinations)
 
 
+class PolicyCredential:
+    """
+    A portable structure that contains information necessary for Alice or Bob
+    to utilize the policy on the network that the credential describes.
+    """
+
+    def __init__(self, alice_verifying_key, label, expiration, policy_pubkey,
+                 treasure_map=None):
+        self.alice_verifying_key = alice_verifying_key
+        self.label = label
+        self.expiration = expiration
+        self.policy_pubkey = policy_pubkey
+        self.treasure_map = treasure_map
+
+    def to_json(self):
+        """
+        Serializes the PolicyCredential to JSON.
+        """
+        cred_dict = {
+            'alice_verifying_key': bytes(self.alice_verifying_key).hex(),
+            'label': self.label.hex(),
+            'expiration': self.expiration.iso8601(),
+            'policy_pubkey': bytes(self.policy_pubkey).hex()
+        }
+
+        if self.treasure_map is not None:
+            cred_dict['treasure_map'] = bytes(self.treasure_map).hex()
+
+        return json.dumps(cred_dict)
+
+    @classmethod
+    def from_json(cls, data: str):
+        """
+        Deserializes the PolicyCredential from JSON.
+        """
+        cred_json = json.loads(data)
+
+        alice_verifying_key = UmbralPublicKey.from_bytes(
+                                    cred_json['alice_verifying_key'],
+                                    decoder=bytes().fromhex)
```

Comment:
Extra set of parenthesis after `bytes` here - passing the `fromhex` method bound to an empty bytestring.

Reply:
Not an extra set, this is intentional. :)

Reply:
@tuxxy - Why do this / What effect does it have? Why not use the class method directly `bytes.from_hex`?

Reply:
Yeah, I think `bytes.fromhex` works just as well